lang/syn/src/codegen/program/handlers.rs

```diff
@@ -356,6 +379,8 @@ pub fn generate(program: &Program) -> proc_macro2::TokenStream {
                                     accounts: &[AccountInfo],
                                     ix_data: &[u8],
                                 ) -> ProgramResult {
+                                    #[cfg(not(feature = "no-log-ix-name"))]
+                                    anchor_lang::prelude::msg!(format!("Instruction: {}", #ix_name_log));
```

Comment:
Historically `format!` strings have been quite expensive. Not sure about this particular case, but we should probably format the string at compile time instead of runtime.

Reply:
yup, good catch. wasn't intentional. ix_name_log is already formatted just forgot to remove it in this case

Reply:
It's so cool that we can do such things at compile time :tada: